File: src/main/scala/com/dolly/wc/StreamWordCount.scala
package com.dolly.wc

import org.apache.flink.streaming.api.scala._

/**
 * 流处理word count程序
 *
 * @author yusenyang 
 * @create 2020/8/1 18:20
 */
object StreamWordCount {

  def main(args: Array[String]): Unit = {
    // 创建一个执行环境
    val env = StreamExecutionEnvironment.getExecutionEnvironment

    // 接收一个socket文本流
    val dataStream = env.socketTextStream("localhost", 7777)

    // 对每条数据进行处理
    val wordCountDataStream = dataStream
      .flatMap(_.split(" "))
      .filter(_.nonEmpty)
      .map((_,1))
      .keyBy(0)
      .sum(1)

    // 打印结果
    wordCountDataStream.print()

    // 启动executor
    // nc -lk 7777
    env.execute("stream word count job")
  }
}

File: src/main/java/com/github/dolly0526/flink/demo/StreamWordCount.java
package com.github.dolly0526.flink.demo;

import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * @author yusenyang
 * @create 2021/3/16 14:25
 */
public class StreamWordCount {

    public static final String host = "localhost";
    public static final int port = 7777;


    public static void main(String[] args) throws Exception {

        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        DataStream<String> inputDataStream = env.socketTextStream(host, port);

        DataStream<Tuple2<String, Integer>> wordCountDataStream = inputDataStream
                .flatMap(new MyFlatMapper())
                .keyBy(0)
                .sum(1);

        wordCountDataStream.print().setParallelism(1);

        env.execute();
    }
}
